programs/stake/src/stake_state.rs

```diff
@@ -1152,6 +1172,87 @@ impl<'a> StakeAccount for KeyedAccount<'a> {
     }
 }
 
+#[derive(Clone, Debug, PartialEq)]
+enum MergeKind {
```

Comment:
I've factored most of this change set's logic under this guy.  Hopefully it's a little better self-documenting now